### PR TITLE
cleanup debug code, fix uip TIME_WAIT, changes in httplog, add compiler options

### PR DESCRIPTION
**Pull request by ferbar**
cleanup debug code (in uip, 
fix uip TIME_WAIT (no need to wait for 30 seconds to timeout)
chanes in httplog (fix memory leak, store uuip in eeprom, add ecmd commands, save some bytes ram)
add compiler options (to create a .map and .lst files)
DNS: save some bytes ram

### CODE REVIEW

File: config.in [hunk 1]
```diff
@@ -43,6 +43,10 @@ if [ "$ARCH_AVR" = "y" ]; then
 	choice 'Hardware/Periphery Class'	\
 			"$(scripts/pinning_hardware_list.sh)" \
 		'Etherrape' HARDWARE
+		
+	bool 'create map file' DEBUG_MAPFILE
+	bool 'create list file' DEBUG_LISTFILE
+	bool 'GCC warnings' GCC_VERBOSE
```
Comment (michaelb42) [inline config.in]: Not needed. make VERBOSE=y does the trick.

File: doc/Configure.help [hunk 2]
```diff
@@ -3589,3 +3594,13 @@ DEBUG_HD44780_CHARCONV
...
+
+DEBUG_MAPFILE
+  Use this option to find strings which can be converted to PSTR() const variables.
+  Have a look at .rodata in the generated ethersex.map file.
+
+DEBUG_LISTFILE
+  Use this option to find the function where the const data can be converted to PSTR()
+
+GCC_VERBOSE
+  Enable many gcc warnings
```
Comment (michaelb42) [inline doc/Configure.help]: Not needed. See above.

File: scripts/defaults.mk [hunk 3]
```diff
@@ -44,10 +44,10 @@ endif # MAKECMDGOALS!=menuconfig
 endif # MAKECMDGOALS!=mrproper
 endif # MAKECMDGOALS!=clean
 
-ifneq ($(VERBOSE),y)
-CFLAGS ?= -Wall -W -Wno-unused-parameter -Wno-sign-compare -Wno-char-subscripts
+ifneq ($(GCC_VERBOSE),y)
+CFLAGS += -Werror=implicit-function-declaration -Wuninitialized -Wformat -Wall -W -Wno-unused-parameter -Wno-sign-compare -Wno-char-subscripts
```
Comment (michaelb42) [inline scripts/defaults.mk]: ?= and += are not the same. ?= was used on purpose here.

File: scripts/defaults.mk [hunk 4]
```diff
@@ -44,10 +44,10 @@ endif # MAKECMDGOALS!=menuconfig
...
 else
-CFLAGS ?= -Wall -W -Wno-unused-parameter -Wshadow -Wpointer-arith -Wno-cast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Winline -Wbad-function-cast -Wsign-compare -Wnested-externs
+CFLAGS += -Werror=implicit-function-declaration -Wuninitialized -Wformat -Wall -W -Wno-unused-parameter -Wshadow -Wpointer-arith -Wno-cast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Winline -Wbad-function-cast -Wsign-compare -Wnested-externs
```
Comment (michaelb42) [inline scripts/defaults.mk]: See above.

File: scripts/defaults.mk [hunk 5]
```diff
@@ -44,10 +44,10 @@ endif # MAKECMDGOALS!=menuconfig
...
 M4FLAGS ?= --debug=V
```
Comment (michaelb42) [inline scripts/defaults.mk]: surprise. surprise. m4 will spill lots of debug output if GCC_VERBOSE is set, too? nope, don't like that.

File: Makefile [hunk 6]
```diff
@@ -21,6 +24,13 @@ ifneq ($(MAKECMDGOALS),indent)
 
 include $(TOPDIR)/.config
 
+ifdef DEBUG_MAPFILE
```
Comment (ferbar) [inline Makefile]: where do you want this code?

**Comment by michaelb42** [comment]
Closed. Author is not willing to split up the PR as requested. Also ref #485

**Comment by eku** [comment]
Replaced by #547.

**Pull request closed without merging**